tests/socket.cpp

```diff
@@ -27,3 +30,43 @@ TEST_CASE("socket sends and receives const buffer", "[socket]")
     CHECK(2 == receiver.recv(buf, 2));
     CHECK(0 == memcmp(buf, "Hi", 2));
 }
+
+#ifdef ZMQ_CPP11
+TEST_CASE("socket proxy", "[socket]")
+{
+    zmq::context_t context;
+    zmq::socket_t front(context, ZMQ_ROUTER);
+    zmq::socket_t back(context, ZMQ_ROUTER);
+    zmq::socket_t capture(context, ZMQ_DEALER);
+    front.bind("inproc://test1");
+    back.bind("inproc://test2");
+    capture.bind("inproc://test3");
+    auto f = std::async(std::launch::async, [&]() {
+        auto s1 = std::move(front);
+        auto s2 = std::move(back);
+        auto s3 = std::move(capture);
+        CHECK_THROWS(zmq::proxy(s1, s2, &s3));
```

Comment:
Does this throw because of an ETERM error because the context is closed?

Shouldn't the CHECK_THROW be placed where the future is retrieved with get?

Reply:
Added explicit check for ETERM (constant available on windows?) and moved the check.